integrationtest/legacy_client/main.d

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+
+    Test that legacy plugins compile with client core
+
+    copyright:
+        Copyright (c) 2018 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module integrationtest.legacy_client.main;
+
+import swarm.client.model.IClient;
+import swarm.util.ExtensibleClass;
+import swarm.client.plugins.ScopeRequests;
+
+// Check that the client and plugins compile.
+class Client : IClient
+{
+    import swarm.client.request.notifier.IRequestNotification;
+
+    mixin ExtensibleClass!(ScopeRequestsPlugin);
+
+    // Required by ScopeRequests (don't ask me why)
+    alias IRequestNotification RequestNotification;
+
+    this ( )
+    {
+        super(null, null);
+    }
+
+    override protected void scopeRequestParams (
+        void delegate ( IRequestParams params ) dg ) { }
+}
+
+version (UnitTest){}
+else
+void main ( )
+{
+}
```

Comment:
Why can't this be a unittest ?

Reply:
It felt nicer to put this in a separate module, as it's testing bringing together code from several other modules (the client base class and the plugins). It would have felt slightly odd to add such a test in any of those other modules. (Though I don't mind moving it, if you absolutely object.)

Reply:
I was missing the fact that it's bringing multiple modules together. Makes perfect sense then!

Reply:
Actually, I would suggest making it a unittest in a separate module with `_test.d` suffix. I think this is more fitting for "lot of modules but no environment/state" kind of tests.

Reply:
Right, that just occurred to me too.

Reply:
Updated.